### PR TITLE
Add versioning options

### PR DESCRIPTION
**Pull request by vtbassmatt**
Adds a few versioning options that weren't present before.

### CODE REVIEW

**Comment by codepic** [comment]
@vtbassmatt ok, coming from TFS world and trying my best here, how should I best pull this version to my fork? Or should I just delete my fork and start from this pull request (which I don't know how). 

Anyways, all I want to do is to pull your changes on this specific pull request and test it on our environment.

**Comment by vtbassmatt** [comment]
You can get my changes by:

```winbatch
git checkout master
git pull
git checkout users/macoope/packversion
```

You don't need to delete your fork to play with mine. We can worry about integrating your changes (or starting over) as needed.

To upload the newer build task, you'll need the [tfx](https://www.npmjs.com/package/tfx-cli) utility.

- Get a [PAT](http://blog.devmatter.com/personal-access-tokens-vsts/) for your account
- `tfx login --service-url https://{ACCOUNT}.visualstudio.com/DefaultCollection --token {PAT}`
- cd into `vso-agent-tasks\Tasks\NugetInstaller`
- `git checkout users/macoope/packversion` (if you didn't above)
- `tfx build tasks upload --task-path . --overwrite`

File: Tasks/NugetPackager/task.json [hunk 1]
```diff
@@ -46,6 +43,61 @@
...
+            "name": "versionByBuild",
+            "type": "pickList",
+            "label": "Automatic package versioning",
+            "defaultValue":"false",
+            "helpMarkDown": "If you choose 'Use semantic versioning', this will generate a [SemVer](http://semver.org/spec/v1.0.0.html)-compliant version formatted as `X.Y.Z-ci-datetime` where you choose X, Y, and Z.\n\nIf you choose 'Use an environment variable', you must select an environment variable and ensure it contains the version number you want to use.\n\nIf you choose 'Use the build number', this will use the build number to version you package. **Note:** Under General set the build format to be '$(BuildDefinitionName)_$(Year:yyyy).$(Month).$(DayOfMonth)$(Rev:.r)'.",
+            "required":true,
+            "groupName": "versioning",
+            "options": {
+                "false": "Off",
+                "byPrereleaseNumber": "Use semantic versioning",
```
Comment (jmyersmsft) [inline Tasks/NugetPackager/task.json]: This choice doesn't really have anything to do with semantic versioning. It's a bit like answering "I use Windows" when asked "What web browser do you use?"

Comment (vtbassmatt) [inline Tasks/NugetPackager/task.json]: Open to suggestions. Before I had said "generate a prerelease version string" which was pretty terrible but arguably more correct.

**Pull request merged**